camera_calibration_parsers/test/test_parse_yml.cpp

```diff
@@ -175,3 +175,62 @@ TEST(ParseYml, roundtrip_calib8) {
   ASSERT_EQ(camera_name2, camera_name);
   check_calib(cam_info2);
 }
+
+static const char * kValidCalibRoi =
+  R"(
+image_width: 640
+image_height: 480
+camera_name: mono_left
+camera_matrix:
+  rows: 3
+  cols: 3
+  data: [1, 2, 3, 4, 5, 6, 7, 8, 9]
+distortion_model: plumb_bob
+distortion_coefficients:
+  rows: 1
+  cols: 5
+  data: [1, 2, 3, 4, 5]
+rectification_matrix:
+  rows: 3
+  cols: 3
+  data: [1, 0, 0, 0, 1, 0, 0, 0, 1]
+projection_matrix:
+  rows: 3
+  cols: 4
+  data: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]
+binning_x: 1
+binning_y: 2
+roi:
+  width: 600
+  height: 300
+  x_offset: 20
+  y_offset: 180
+  do_rectify: true
```

Comment:
You should also make tests for when one or more of these values is not specified to verify error behaviour.

Reply:
Currently all plausibility checks are in the consumer nodes of CameraInfo messages. So it I'm not sure how the error behaviour should look like. Currently, it fails with an error message when the `roi` YAML node is present but one of its children isn't. I think it is a meaningful behavior since it ensures the user does not accidentally forget one of these fields. Should I add really add a test for each of the `roi` fields and their combinations? When I also add binning then these would be 2^8 (=lines) possible tests.

Reply:
OK, that sounds fine.